koku/api/iam/view/user_preference.py

```diff
@@ -0,0 +1,272 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""View for User Preferences."""
+
+from rest_framework import exceptions, mixins, viewsets
+from rest_framework.authentication import (SessionAuthentication,
+                                           TokenAuthentication)
+from rest_framework.permissions import IsAuthenticated
+
+import api.iam.models as models
+import api.iam.serializers as serializers
+from api.common.permissions.owner import IsOwner
+
+
```

Comment:
Not exactly as was described here: https://github.com/project-koku/koku/issues/97

But I am ok with it. May have misunderstood, but I assumed user -> user preference would be one-to-one. 

Was thinking a `PATCH` against `api/v1/users/(user_id)/preferences/` and a `GET` on `api/v1/users/(user_id)/preferences/`

Reply:
Isn't User -> Preference a one-to-many relationship because we're expecting to set preference entries on a per-topic basis? e.g. one row for currency prefs, one row for timezone prefs.

I messaged you and that's what you said when we discussed it last.

Reply:
Like I said I'm ok with this. I think it was a lost in translation interpretation of what you were meaning vs. what I thought you meant. Definitely not asking you to do a big rewrite here.